### PR TITLE
Rework About screen typography

### PR DESCRIPTION
**Pull request by seadowg**
This is blocked by #3217.

Here I used the Material Guidelines [Lists spec](https://material.io/design/components/lists.html#specs) as a jumping off point for the text scales.

The reasoning behind what we're doing is [here](https://forum.opendatakit.org/t/reworking-collects-typography/20634). I've also, while in the area, tried to clean up the layouts involved by reworking the structure, using standard values and extracting common styles.

Here is before the changes:

Screenshot_1562848230

And after:

Screenshot_1562853002

#### What has been done to verify that this works as intended?

Ran tests, played around with effected screen.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/res/values/attrs.xml
Comment (seadowg) [inline collect_app/src/main/res/values/attrs.xml:12]: This was an interesting one: As Android doesn't have a way to change emphasis/opacity on `TextView`s without changing the full color it made sense to define a medium emphasis version of the standard `colorOnSurface` (what would be the base color for text on the white surface) attribute. The value (set in `theme.xml`) is set to 60% which corresponds with the working [color palette](https://github.com/opendatakit/collect/files/3382372/collect-styleguide.pdf).

**Comment by yanokwa** [comment]
Nitpick: Sad that About doesn't line up with Visit/Join/Share. Can that be improved easily?

**Comment by seadowg** [comment]
@yanokwa hmmm that's a good point. I'm (as far as I know) following the Material guideline margins here so it could be that I've missed something or the App bar title is not where Material expects it to be. I'll take a look!

**Comment by seadowg** [comment]
@yanokwa yeah looks like our margin on the App Bar isn't the latest, greatest, officially signed off Material App Bar margin: https://material.io/design/components/app-bars-top.html#specs.

I'll have a look at what changing that would look like for us. We can block this for the moment!

P.S. Why can't I re-draft a PR? That would be the best.

**Comment by yanokwa** [comment]
It's your call on whether or not we need to block. The items don't line up in master, so it's not a regression. But if it's easy to add, we might want to block so we do one big QA pass.

I too hate not being able to undraft. I'll talk to my connects at GitHub and let them know. In the meantime, we can prepend WIP, like animals.

**Comment by seadowg** [comment]
It looks like we were customizing the margin on the App Bar so I have removed that. This means every screen now has more of an inset before the App Bar title but I think it's a good change as the closer we get towards Material Guidelines the more that will actually all line up. Plus, it means as they iterate on the App Bar we won't overriding those margins.

Here's a screenshot:

Screenshot_1562862609

**Comment by seadowg** [comment]
@ping @grzesiek2010 would be cool to get your thoughts on this

File: collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml
Comment (lognaturel) [inline collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml:1]: I would tend to leave a comment with some of the context at https://github.com/opendatakit/collect/pull/3207/commits/a81dd497e98d90fbcc647299d9c1921859c1fde3#r302561112? I might be an over-commenter, though.

**Review by lognaturel — DISMISSED** [review]

File: collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml
Comment (seadowg) [reply collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml:1]: Do you mean a comment in the code itself or as part of the PR? I actually feel like I'm probably under commenting so good to hear feedback when you think things need some comments - always working with teams all in one place is a good way to think nothing needs comments.

Comment (lognaturel) [reply collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml:1]: I mean as an XML comment. They're allowed there, right?

Comment (seadowg) [reply collect_app/src/main/res/color/color_on_surface_medium_emphasis.xml:1]: Yeah totally. I can look at adding some comment to this!

**Review by grzesiek2010 — DISMISSED** [review]

**Comment by mmarciniak90** [comment]
@seadowg I noticed that scroll bar is not visible

**Comment by seadowg** [comment]
@mmarciniak90 hmm that's a good catch. I'm seeing that in the 1.23 beta as well though? Looks like this might be an existing issue - could you confirm?

**Comment by seadowg** [comment]
This is blocked by #3217

**Comment by mmarciniak90** [comment]
@seadowg you have right, it is an old problem.
I create a separate issue for it - #3226 

**Tested with success**

**Tested on Android**: 8.1, 7.0, 6.0, 5.1, 4.4, 4.2, 4.1 (because of the size of the screen)

|1.23.0 beta2 |version with fix|
|---|---|
|Screenshot_20190722-075447|Screenshot_20190722-075345|

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Comment by lognaturel** [comment]
This was in a confusing state because it looked like it has two approvals and has gone through QA but isn't merged. The reason for that is that it was rebased on top of #3217 so action needs to be taken there first. I've dismissed the reviews to make that clear.

**Comment by lognaturel** [comment]
It'd be helpful to get this synced up with master before the final check.

**Comment by seadowg** [comment]
@lognaturel because this is blocked by #3217 I think it makes sense to wait for that to merge before rebasing otherwise we get new commit SHAs for the changes from that PR. I'll make it [WIP] right now to make that clear. In future I'll just keep PRs like this as a draft (like I did with #3225).

**Comment by lognaturel** [comment]
Oh, yes, absolutely. Apologies, I got ahead of myself or something -- I remembered merging #3217 but in fact we'll do a QA pass.

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
@mmarciniak90 hopefully you'll get the same results as in https://github.com/opendatakit/collect/pull/3207#issuecomment-513652109!

**Comment by mmarciniak90** [comment]
**Tested with success**

Tested on Android: 9.0, 8.1, 7.0, 6.0, 5.1, 4.4, 4.2

Verified cases:
* multi clicking on options
* rotating screen
* scrolling 
* Spanish and English language 
* light/dark mode

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**
